staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go

```diff
@@ -97,8 +106,33 @@ type CSINodeInfo struct {
 	CSIDrivers []CSIDriverInfo `json:"csiDrivers" patchStrategy:"merge" patchMergeKey:"driver"`
 }
 
-// CSIDriverInfo contains information about one CSI driver installed on a node.
+// CSIDriverInfo holds information about one CSI driver installed on a node.
 type CSIDriverInfo struct {
+	// spec is the specification of CSIDriverInfo
+	Spec CSIDriverInfoSpec `json:"spec"`
+
+	// status is the current status of CSIDriverInfo
+	Status CSIDriverInfoStatus `json:"status"`
+}
+
+// CSIDriverInfoStatus holds information about the status of one CSI driver installed on a node
+type CSIDriverInfoStatus struct {
+	// available is a boolean representing whether the driver has been installed
+	// on this node or not.
+	Available bool `json:"available"`
+
+	// volumePluginMechanism announces what mechanism underlies volume plugins.
+	// It is set by Kubelet. It is used by the attach/detach controller, which
+	// needs to know how to perform attachments. The allowed values are:
+	// * "in-tree": the volume operation (e.g., attach/detach) ought to be
+	//   directly performed by the attach/detach controller.
+	// * "csi-plugin": the attach/detach controller ought to request
+	//   the csi plugin to perform the volume operation rather than perform it directly.
+	VolumePluginMechanism VolumePluginMechanism `json:"volumePluginMechanism"`
```

Comment:
Do you have a webhook for validating this? What happens if this is set to an invalid value?

Reply:
possible to just use CRD validation?

```
volumePluginMechanism:
                    description: Indicates to external components the required mechanism
                      to use for any in-tree plugins replaced by this driver.
                    pattern: in-tree|csi
                    type: string
```

Reply:
on validation failure: `csiDrivers.status.volumePluginMechanism in body should match 'in-tree|csi'`